src/proxy/outbound.rs

```diff
@@ -329,6 +278,14 @@ impl OutboundConnection {
             .body(())
             .expect("builder with known status code should not fail");
 
+        let pool_key = Box::new(pool::WorkloadKey {
+            src_id: req.source.identity(),
+            // Clone here shouldn't be needed ideally, we could just take ownership of Request.
```

Comment:
Yeah - why construct the poolkey here at all? We can just chuck the whole request + remote_addr at `send_request_pooled`.

Reply:
We have the same clone problem either way I think (actually, _maybe_ not since we Box::pin the future).

If we pass owned `Request` here we can easy deconstruct the request and not clone `upstream_sans`. But then we end up paying for the stack cost of Request twice (and I think it somehow actually becomes 4x due to some async stuff?). If we have a reference we don't need to, but then we need to clone this. If we kick this down further into `send_request_pooled` we have the same problem, just possibly more stack usage I think?

I guess maybe an owned `Box<Request>` wouldn't have that problem...

Reply:
> I guess maybe an owned Box<Request> wouldn't have that problem...

That would be my intuition, but IDK for sure. Eh, it's a small thing either way.